tools/syz-declextract/syz-declextract.cpp

```diff
@@ -1,279 +1,93 @@
-      if (strncmp(fn.c_str(), prefix, strlen(prefix)) == 0) return true;
-    }
-    const char *ignore_functions[] {
-      "IEnum",
-      "IStream",
-      "IType",
-      "IService",
-      "IProperty",
-      "ISequential",
-      "IDispatch",
-      "I_RPC",
-      "I_Rpc",
-      "CLEANLOCAL",
-      "WinMain",
-      "PropertySheet",
-      "LookupAccountNameLocalA",
-      "LookupAccountSidLocalA",
-      "WTSGetServiceSessionId",
-      "WTSIsServerContainer",
-      "GetDisplayAutoRotationPreferencesByProcessId",
-      "LoadStringByReference",
-      "IdnToNameprepUnicode",
-      "VerFindFileA",
-      "VerInstallFileA",
-      "GetFileVersionInfoSizeA",
-      "GetFileVersionInfoA",
-      "GetFileVersionInfoSizeExA",
-      "GetFileVersionInfoExA",
-      "VerQueryValueA",
-      "sndOpenSound",
-      "Netbios",
-      "RpcBindingGetTrainingContextHandle",
-      "RpcAsyncCleanupThread",
-      "ShellMessageBoxA",
-      "SHEnumerateUnreadMailAccountsA",
-      "SHGetUnreadMailCountA",
-      "SHSetUnreadMailCountA",
-      "GetEncSChannel",
-      "CryptExportPKCS8Ex",
-      "FindCertsByIssuer",
-      "CryptCancelAsyncRetrieval",
-      "CryptGetTimeValidObject",
-      "CryptFlushTimeValidObject",
-      "CryptProtectDataNoUI",
-      "CryptUnprotectDataNoUI",
-      "NsServerBindSearch",
-      "NsClientBindSearch",
-      "NsClientBindDone",
-      "GetOpenCardNameA",
-      "SubscribeServiceChangeNotifications",
-      "UnsubscribeServiceChangeNotifications",
-      "GetThreadDescription",
-      "SetThreadDescription",
-      "DialogControlDpi",
-      "SetDialogDpiChangeBehavior",
-      "GetDialogDpiChangeBehavior",
-      "RpcServer",
-      "DecodePointer",
-      "DecodeRemotePointer",
-      "DecodeSystemPointer",
-      "EncodePointer",
-      "EncodeRemotePointer",
-      "EncodeSystemPointer",
-      "UnmapViewOfFile2",
-      "MapViewOfFileNuma2",
-      "DeriveCapabilitySidsFromName",
-      "QueryAuxiliaryCounterFrequency",
-      "ConvertPerformanceCounterToAuxiliaryCounter",
-      "ConvertAuxiliaryCounterToPerformanceCounter",
-      "FreePropVariantArray",
-      "PropVariantCopy",
-      "PropVariantClear",
-      "InitiateShutdown",
-      "ExitWindowsEx",
-      "LockWorkStation",
-      "InitiateSystemShutdown",
-      "InitiateSystemShutdownEx",
-      "shutdown",
-    };
-    for (auto func: ignore_functions) {
-      if (strstr(fn.c_str(), func)) return true;
-    }
-    // These are already described:
-    const char *ignore_exact[] {
-      "CreateFileA",
-      "CloseHandle",
-      "VirtualAlloc",
-    };
-    for (auto func: ignore_exact) {
-      if (strcmp(fn.c_str(), func) == 0) return true;
+class Printer : public MatchFinder::MatchCallback {
+public:
+  virtual void run(const MatchFinder::MatchResult &Result) override {
+    const auto *varDecl = Result.Nodes.getNodeAs<VarDecl>("Struct");
+    auto *context = Result.Context;
+    if (!varDecl || !varDecl->getInit()) {
```

Comment:
Both [LLVM](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements) and [Google](https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching) style guides suggest that you don't need curly braces around a single return statement here (and below):

```
if (!varDecl || !varDecl->getInit())
  return;
```

Reply:
Done!